druntime/src/core/internal/gc/impl/conservative/gc.d

```diff
@@ -3600,21 +3569,25 @@ Lmark:
         }
         assert(pbot < ptop);
 
-        busyThreads.atomicOp!"+="(1); // main thread is busy
-
-        evStart.setIfInitialized();
+        evStackFilled.setIfInitialized(); // background threads start now
 
         debug(PARALLEL_PRINTF) printf("mark %lld roots\n", cast(ulong)(ptop - pbot));
 
+        void pullLoop(bool precise)()
+        {
+            static if (precise)
+                mark!(true, true, true)(ScanRange!true(pbot, ptop, null));
+            else
+                mark!(false, true, true)(ScanRange!false(pbot, ptop));
+
+            while (pullFromScanStack())
```

Comment:
```suggestion
            while (pullFromScanStackImpl!precise())
```
Don't pay the price of checking the runtime boolean again.

Reply:
Good catch, updated.